.github/workflows/e2e-tests-main-devnet.yml

```diff
@@ -532,6 +532,148 @@ jobs:
           ALEPH_NODE_BINARY: aleph-release-node/aleph-node
         run: ./.github/scripts/test_multiple_restarts.sh
 
+  check-runtime-change:
+    name: Inspect whether runtime version has been changed (comparing to main)
+    runs-on: ubuntu-latest
+    env:
+      OLD_REPO_PATH: old-repo
+      NEW_REPO_PATH: new-repo
+      OUTPUT_NAME: diff
+    outputs:
+      runtime-updated: ${{ steps.inspect.outputs.diff }}
+    steps:
+      - name: Checkout old source code (main branch)
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+          path: ${{ env.OLD_REPO_PATH }}
+
+      - name: Checkout current source code
+        uses: actions/checkout@v3
+        with:
+          path: ${{ env.NEW_REPO_PATH }}
+
+      - name: Inspect versions
+        id: inspect
+        run: |
+          ./${{ env.NEW_REPO_PATH }}/.github/scripts/check_runtime_change.sh \
+              ${{ env.OLD_REPO_PATH }} \
+              ${{ env.NEW_REPO_PATH }} \
+              ${{ env.OUTPUT_NAME }}
+
+  build-old-node:
+    name: Build node artifact (main version)
+    needs: [ check-runtime-change ]
+    if: ${{ needs.check-runtime-change.outputs.runtime-updated != 0 }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+
+      - name: Install WASM target
+        run: rustup target add wasm32-unknown-unknown
+
+      - name: Restore cache
+        uses: ./.github/actions/restore-cache
+        with:
+          target-key: try-runtime
+          cargo-key: old
+
+      - name: Build old binary
```

Comment:
There're two approaches to this feature:
1) build the binary from `main` as a baseline for a runtime upgrade
2) use Testnet as such
Now they obviously test different things, but I think what we want to test here is the latter. This is because we want to be prepared for upcoming Testnet updates, so essentialy test runtime upgrade from version `X` to `Y`, where `X` is top of `main`, and `Y` is Testnet one

Reply:
done, but I'm a little worried about storage scraping time... we'll see